spec/compiler/semantic/def_spec.cr

```diff
@@ -171,6 +171,16 @@ describe "Semantic: def" do
       "undefined method"
   end
 
+  it "errors when type restriction is incompatible with default value" do
```

Comment:
I'd rephrase it: `errors when default value is incompatible with type restriction`